packages/library-base/src/commonMain/kotlin/io/realm/internal/query/ScalarQuery.kt

```diff
@@ -0,0 +1,131 @@
+    realmReference: RealmReference,
+    queryPointer: NativePointer,
+    mediator: Mediator
+) : BaseScalarQuery<Long>(realmReference, queryPointer, mediator) {
+
+    override fun find(): Long = RealmInterop.realm_query_count(queryPointer)
+
+    override fun getScalarClass(): KClass<Long> = Long::class
+
+    override fun Flow<BaseResults<Long>?>.queryMapper(): Flow<Long> = TODO()
+
+    override fun discardRepeated(latestValue: Long?): Boolean = TODO()
+}
+
+@Suppress("LongParameterList")
+internal class AggregatorQuery<T : Any> constructor(
+    realmReference: RealmReference,
+    queryPointer: NativePointer,
+    mediator: Mediator,
+    private val clazz: KClass<*>,
+    private val property: String,
+    private val type: KClass<T>,
+    private val queryType: AggregatorQueryType
+) : BaseScalarQuery<T>(realmReference, queryPointer, mediator) {
+
+    override fun find(): T? = findInternal(queryPointer)
+
+    override fun getScalarClass(): KClass<T> = type
+
+    override fun Flow<BaseResults<T>?>.queryMapper(): Flow<T?> = TODO()
+
+    override fun discardRepeated(latestValue: T?): Boolean = TODO()
+
+    private fun findInternal(queryPointer: NativePointer): T? =
+        findFromResults(RealmInterop.realm_query_find_all(queryPointer))
+
+    private fun findFromResults(resultsPointer: NativePointer): T? = try {
+        val colKey = getPropertyKey(clazz, property).key
+        computeAggregatedValue(resultsPointer, colKey)
+    } catch (exception: RealmCoreException) {
+        throw when (exception) {
+            is RealmCoreLogicException ->
+                IllegalArgumentException(
+                    "Invalid query formulation: ${exception.message}",
+                    exception
+                )
+            else ->
+                genericRealmCoreExceptionHandler(
+                    "Invalid query formulation: ${exception.message}",
+                    exception
+                )
+        }
+    }
+
+    private fun computeAggregatedValue(resultsPointer: NativePointer, colKey: Long): T? {
+        val result: T? = when (queryType) {
+            AggregatorQueryType.MIN ->
+                RealmInterop.realm_results_min(resultsPointer, colKey)
+            AggregatorQueryType.MAX ->
+                RealmInterop.realm_results_max(resultsPointer, colKey)
+            AggregatorQueryType.SUM ->
+                RealmInterop.realm_results_sum(resultsPointer, colKey)
+        }
+
+        val numberResult: Number = result?.let { it as Number } ?: return null
+
+        // TODO Expand to support other numeric types, e.g. Decimal128
+        return when (type) {
+            Int::class -> numberResult.toInt()
+            Long::class -> numberResult.toLong()
+            Float::class -> numberResult.toFloat()
+            Double::class -> numberResult.toDouble()
+            else -> throw IllegalArgumentException("Invalid numeric type for '$property', it is not a '${type.simpleName}'.")
```

Comment:
Should we also allow conversion to strings or does that just raise too many questions about format so better left up to developers?

Reply:
We should probably give people a list of supported types here.

Reply:
We also need to support `RealmInstant` for min and max...`sum` probably doesn't make sense.

Reply:
I suspect adding string conversion would make it a bit more confusing, so I would leave it as is.